### PR TITLE
Upgrade to Composer 2 / Airflow 2

### PR DESCRIPTION
**Pull request by charlielewisme**
## What?
- Upgrade to GCP Composer 2
- Upgrade to Airflow 2

## Why?
Composer 1 is being deprecated at end of March 2023

## How? 
- [x] Add airflow import test and workflow, update other workflows etc.
- [x] Update and pin requirements
- [x] Airflow 2 code changes
  - [x] Jinja macros
    - [x] execution_date -> logical_date
  - [x] Imports
    - [x] BashOperator
    - [x] EmailOperator
    - [x] PythonOperator
    - [x] ExternalTaskSensor
    - [x] PostgresHook
  - [x] Hooks and Operators usage
    - [x] PythonOperator delete provide_context
    - [x] GoogleCloudStorageHook -> GCSHook
    - [x] BigQueryOperator -> BigQueryInsertJobOperator
- [x] Fix local storage limits by moving to GCS Fuse for Composer
- [x] Build Composer dev env using update scripts (shared privately)
- [x] Update local Airflow instance (inc. removing auth requirement)
- [x] Update docs for Composer 2 build

## Testing
- Build dev environment
- Encounter previously seen issue where Composer's install of PyPi packages throws unexpected conflicts which can't be replicated in local installs
  - Fix by pinning conflicting dependency `eth-hash`
- Encounter undocumented issue where scheduler fails to be given access to the whole dags folder by GCS-Fuse when there are a large number of files.
  - Fix by increased Scheduler RAM from 2 to 4GB
- Notice slight change to `airflow_monitoring.py` in latest Composer
  - Update the file in repo
- Run `ethereum_export_dag`: all tasks succeed
- Run `ethereum_load_dag`: all tasks succeed except verifying blocks count (due to lack of backfill)
  - In BigQuery I diffed all tables in `crypto_ethereum` except 'balances', over 2022-09-12 -> 2022-09-14. Everything matched.
- Run `ethereum_parse_common_dag`: tasks started running fine but fail (due to lack of internal dev project)
- Encounter well-known issue where workers sometimes fail to write logs to the logs folder in Composer bucket
  - This is not fixable with `log_fetch_timeout_sec` since this config has been [blocked in Composer 2.2.5](https://cloud.google.com/composer/docs/concepts/airflow-configurations#blocked_configurations_in_airflow_225)
  - I speculatively increased worker RAM from 4 to 8GB to see if it helps

- In general, further resource tweaks should be expected in production

### CODE REVIEW

File: dags/airflow_monitoring.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     default_args=default_args,
     description='liveness monitoring dag',
     schedule_interval=None,
-    dagrun_timeout=timedelta(minutes=60))
+    dagrun_timeout=timedelta(minutes=20))
```
Comment (TimNooren) [inline dags/airflow_monitoring.py:18]: Should we even be managing this DAG at all? As it's coming from the composer environment itself.

> The liveness DAG is stored in the dags/ folder and visible in the Airflow UI. The frequency and contents of the liveness DAG are immutable and must not be modified. Changes to the liveness DAG do not persist.

https://cloud.google.com/composer/docs/composer-2/monitor-environments

**Review by TimNooren — COMMENTED** [review]
LGTM, just two small comments:)

Regarding:

> Encounter previously seen issue where Composer's install of PyPi packages throws unexpected conflicts which can't be replicated in local installs
> 
> - Fix by pinning conflicting dependency eth-hash

Maybe we can add this as a comment to `requirements.txt`? Since it might not be obvious to new readers. As you mention I could also not reproduce this issue locally (`ethereum-etl==2.0.3` seems to install that exact version of `eth-hash`).

File: dags/airflow_monitoring.py (hunk 1 above)
Comment (charlielewisme) [reply dags/airflow_monitoring.py:18]: I agree with this. I would prefer to delete this dag from this repo... we just have to keep it in the exclusion list in `gsutil rsync` calls, e.g.

```
gsutil -m rsync -x 'airflow_monitoring|.*\.pyc$' -cdr "$LOCAL_DAGS_FOLDER" "$DAG_GCS_PREFIX"
```

**Review by TimNooren — APPROVED** [review]

**Comment by charlielewisme** [comment]
@medvedev1088
Please let me know if there's any issues with your deployment.
If it's okay, should I merge this PR?
Not before weekend, I mean Monday at earliest.

**Comment by charlielewisme** [comment]
Actually more retries on parse tasks than I thought. Yesterday's run:
```
Number of tries -- Freq
1 -- 2977
2 -- 384
3 -- 77
4 -- 10
5 -- 1
```

**Comment by charlielewisme** [comment]
Today's run:
```
Number of tries -- Freq
1 -- 3812 = 98.27% first time success rate
2 -- 64
3 -- 3
4 -- 0
5 -- 0
```

**Comment by charlielewisme** [comment]
Last run:

Number of tries -- Freq
1 -- 3928 = 98.92% first time success rate
2 -- 42
3 -- 1
4 -- 0
5 -- 0

**Comment by charlielewisme** [comment]
1 scheduler (4vCPU, 26GB RAM)

Number of tries -- Freq
1 -- 3912 = 98.81% first time success rate
2 -- 46
3 -- 1
4 -- 0
5 -- 0

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM)

```
Number of tries -- Freq
1 -- 3899 = 98.63% first time success rate
2 -- 54
3 -- 0
4 -- 0
5 -- 0
```

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM),
raised worker_concurrency from 8 to 10

```
Number of tries -- Freq
1 -- 3934 = 98.87% first time success rate
2 -- 42
3 -- 3
4 -- 0
5 -- 0
```

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM),
raised worker_concurrency from 10 to 12,
lowered web server cpu from 2 to 1

```
Number of tries -- Freq
1 -- ? = ? first time success rate
2 -- 
3 -- 
4 -- 
5 --

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM)
environment_size medium
raised worker_concurrency from 12 -> 16

```
Number of tries -- Freq
1 -- ? = ?% first time success rate
2 -- ?
3 -- ?
4 -- ?
5 -- ?
```

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM)
environment_size medium
revert worker_concurrency from 16 -> 12
+ parallelism reduced from 48 -> 42

```
Number of tries -- Freq
1 --  = % first time success rate
2 -- 
3 -- 
4 -- 
5 -- 
```

**Comment by charlielewisme** [comment]
2 schedulers (2vCPU, 13GB RAM)
environment_size medium
worker_concurrency 12
+ parallelism increased from 42 -> 54

```
Number of tries -- Freq
1 --  = % first time success rate
2 -- 
3 -- 
4 -- 
5 -- 
```

**Comment by charlielewisme** [comment]
Reverting to our best config so far:
2 schedulers (2vCPU, 13GB RAM)
environment_size medium
worker_concurrency 12
parallelism back 48

**Comment by charlielewisme** [comment]
Same config...

```
Number of tries -- Freq
1 -- 4053 = 98.69% first time success rate
2 -- 53
3 -- 1
4 -- 0
5 -- 0
```

```
Number of tries -- Freq
1 -- 4067 = 98.98% first time success rate
2 -- 42
3 -- 0
4 -- 0
5 -- 0
```

**Review by medvedev1088 — APPROVED** [review]

**Pull request merged**